test/ExecuteUpdateTest.cpp

```diff
@@ -32,35 +32,41 @@ MATCHER_P(AlwaysFalse, msg, "") {
   return false;
 }
 
-// _____________________________________________________________________________
+// Test the `ExecuteUpdate::executeUpdate` method. These tests run on the
+// default dataset defined in `IndexTestHelpers::makeTestIndex`.
 TEST(ExecuteUpdate, executeUpdate) {
-  auto executeUpdate = [](const std::string& update) {
-    // These tests run on the default dataset defined in
-    // `IndexTestHelpers::makeTestIndex`.
-    QueryExecutionContext* qec = ad_utility::testing::getQec(std::nullopt);
-    const Index& index = qec->getIndex();
-    DeltaTriples deltaTriples{index};
+  QueryExecutionContext* qec = ad_utility::testing::getQec(std::nullopt);
+  const Index& index = qec->getIndex();
+  // Perform the given `update` and store result in given `deltaTriples`.
+  auto expectExecuteUpdateHelper = [&qec, &index](const std::string& update,
+                                                  DeltaTriples& deltaTriples) {
     const auto sharedHandle =
         std::make_shared<ad_utility::CancellationHandle<>>();
     const std::vector<DatasetClause> datasets = {};
     auto pq = SparqlParser::parseQuery(update);
     QueryPlanner qp{qec, sharedHandle};
     const auto qet = qp.createExecutionTree(pq);
     ExecuteUpdate::executeUpdate(index, pq, qet, deltaTriples, sharedHandle);
-    return deltaTriples;
```

Comment:
I believe it would be simpler to just let this lambda return `const DeltaTriples&`, then the code could stay the same.
Also if turned into an actual function, the `[&executeUpdate]` reference would not be necessary

Reply:
Return a reference to a local variable?

Anyway, this is unrelated to this PR and was wrong in the current master (which is the reason why almost all tests failed in the last three commits to the master).

Reply:
const references should keep the object alive for long enough. Non-const references would lead to undefined behaviour. Just the quirks of the C++ language.